pkgs/servers/dnschain/default.nix

```diff
@@ -1,11 +1,51 @@
-{ stdenv, nodePackages }:
-
-# to update dnschain, run npm2nix package.json package.nix, and
-# then add "coffee-script" manually as a dependecy for "dnschain"
-# in package.nix.
+{ stdenv, fetchFromGitHub, callPackage, makeWrapper, openssl }:
 
 let
-  np = nodePackages.override { generated = ./package.nix; self = np; };
-in
+  nodePackages = callPackage (import ../../top-level/node-packages.nix) {
```

Comment:
What is the motivation for the move to this construction instead of taking the package set as a parameter? Does override not work?